src/System.Transactions.Local/src/System/Transactions/TransactionState.cs

```diff
@@ -21,719 +21,144 @@ internal abstract class TransactionState
+        private static TransactionStateVolatileSPC s_transactionStateVolatileSPC;
+        private static TransactionStateSPC s_transactionStateSPC;
+        private static TransactionStateAborted s_transactionStateAborted;
+        private static TransactionStateCommitted s_transactionStateCommitted;
+        private static TransactionStateInDoubt s_transactionStateInDoubt;
+
+        private static TransactionStatePromoted s_transactionStatePromoted;
+        private static TransactionStateNonCommittablePromoted s_transactionStateNonCommittablePromoted;
+        private static TransactionStatePromotedP0Wave s_transactionStatePromotedP0Wave;
+        private static TransactionStatePromotedCommitting s_transactionStatePromotedCommitting;
+        private static TransactionStatePromotedPhase0 s_transactionStatePromotedPhase0;
+        private static TransactionStatePromotedPhase1 s_transactionStatePromotedPhase1;
+        private static TransactionStatePromotedP0Aborting s_transactionStatePromotedP0Aborting;
+        private static TransactionStatePromotedP1Aborting s_transactionStatePromotedP1Aborting;
+        private static TransactionStatePromotedAborted s_transactionStatePromotedAborted;
+        private static TransactionStatePromotedCommitted s_transactionStatePromotedCommitted;
+        private static TransactionStatePromotedIndoubt s_transactionStatePromotedIndoubt;
+
+        private static TransactionStateDelegated s_transactionStateDelegated;
+        private static TransactionStateDelegatedSubordinate s_transactionStateDelegatedSubordinate;
+        private static TransactionStateDelegatedP0Wave s_transactionStateDelegatedP0Wave;
+        private static TransactionStateDelegatedCommitting s_transactionStateDelegatedCommitting;
+        private static TransactionStateDelegatedAborting s_transactionStateDelegatedAborting;
+        private static TransactionStatePSPEOperation s_transactionStatePSPEOperation;
+
+        private static TransactionStateDelegatedNonMSDTC s_transactionStateDelegatedNonMSDTC;
+        private static TransactionStatePromotedNonMSDTCPhase0 s_transactionStatePromotedNonMSDTCPhase0;
+        private static TransactionStatePromotedNonMSDTCVolatilePhase1 s_transactionStatePromotedNonMSDTCVolatilePhase1;
+        private static TransactionStatePromotedNonMSDTCSinglePhaseCommit s_transactionStatePromotedNonMSDTCSinglePhaseCommit;
+        private static TransactionStatePromotedNonMSDTCAborted s_transactionStatePromotedNonMSDTCAborted;
+        private static TransactionStatePromotedNonMSDTCCommitted s_transactionStatePromotedNonMSDTCCommitted;
+        private static TransactionStatePromotedNonMSDTCIndoubt s_transactionStatePromotedNonMSDTCIndoubt;
 
         // Object for synchronizing access to the entire class( avoiding lock( typeof( ... )) )
-        private static object s_classSyncObject;
+        internal static object s_classSyncObject;
```

Comment:
>internal [](start = 8, length = 8)

Since you deleted the ClassSyncObject property, I believe it's clear that ClassSyncObject was only referenced from within this class, so it could have been private, rather than internal. So I think we can leave s_classSyncObject private.